### PR TITLE
EN-1588 Java SDK should have batchWrite method

### PR DESCRIPTION
**Pull request by robert-par**
Pull Request Checklist
======================

Please read this for our PR Culture:
------------------------------------
https://incountry.atlassian.net/wiki/spaces/ED/pages/46858869/Pull+requests+culture+Draft

Work relates to:
----------------
Link(s) to Jira Task/User Story/Bug/Epic:
- https://incountry.atlassian.net/browse/EN-1588

Code Quality (Before submitting PR)
------------
- [x] Ensure code compiles and passes Sonar Lint.
- [x] Do your best to ensure integration tests pass. If they don't then...
- [x] Remove, ignore, or mark pending any brittle tests.
- [x] Remove commented out code.

Target Release
--------------
- [x] Please put the target release vehicle this needs to be bundled with.
- [x] Did you increment the version? What is the new version?

### CODE REVIEW

File: src/main/java/com/incountry/crypto/impl/Crypto.java [hunk 1]
```diff
@@ -30,19 +34,20 @@
...
         this.envId = envId;
     }
 
-    public String encrypt(String plainText) throws GeneralSecurityException, IOException {
+    public Pair<String, Integer> encrypt(String plainText) throws GeneralSecurityException, IOException {
         byte[] clean = plainText.getBytes();
         byte[] salt = generateSalt(SALT_LENGTH);
-        byte[] strong = generateStrongPasswordHash(secret, salt, PBKDF2_ITERATIONS_COUNT, KEY_LENGTH);
+        SecretKey secretKeyObj = getSecret(secretKeysData.getCurrentVersion());
+        byte[] strong = generateStrongPasswordHash(secretKeyObj.getSecret(), salt, PBKDF2_ITERATIONS_COUNT, KEY_LENGTH);
```
Comment (vladimir-grezin) [inline src/main/java/com/incountry/crypto/impl/Crypto.java]: Let's use some more meaningful name, e.g. "encryptionKey" or something like this

File: src/main/java/com/incountry/crypto/impl/Crypto.java [hunk 2]
```diff
@@ -87,46 +92,58 @@ private String decryptUnpacked(byte[] parts) throws GeneralSecurityException {
         return new String(decryptedText);
     }
 
+    private SecretKey getSecret(Integer version){
+        SecretKey secret = null;
+        for (SecretKey item : secretKeysData.getSecrets()) {
+            if (item.getVersion() == version) {
+                secret =  item;
```
Comment (vladimir-grezin) [inline src/main/java/com/incountry/crypto/impl/Crypto.java]: extra space

File: src/test/java/com/incountry/StorageTest.java [hunk 3]
```diff
@@ -72,15 +96,15 @@ public void testWrite() throws GeneralSecurityException, StorageException, IOExc
             JSONObject response = new JSONObject(encrypted);
             String key = response.getString("key");
             String encryptedBody = response.getString("body");
-            String actualBodyStr = crypto.decrypt(encryptedBody);
+            String actualBodyStr = crypto.decrypt(encryptedBody, "0");
```
Comment (vladimir-grezin) [inline src/test/java/com/incountry/StorageTest.java]: Why do we use String for Version here? `SecretKey` considers Version as a Number. And we store it as a Number on PoP. Let's review this

File: src/test/java/com/incountry/StorageTest.java [hunk 4]
```diff
@@ -107,6 +131,32 @@ public void testDelete() throws GeneralSecurityException, StorageException, IOEx
...
+            storage.setHttpAgent(agent);
+            List<Record> records = new ArrayList<>();
+            records.add(new Record(country, key, body, profileKey, rangeKey, key2, key3));
+            storage.batchWrite(country, records);
+
+            String encrypted = agent.getCallBody();
+            String keyHash = crypto.createKeyHash(key);
+            JSONArray responseList = new JSONArray(encrypted);
+            for (Object item : responseList) {
+                System.out.println();
```
Comment (vladimir-grezin) [inline src/test/java/com/incountry/StorageTest.java]: Do we need this output?

**Comment by vladimir-grezin** [comment]
Also I'd consider to use another naming convention. Let's try to get rid of `IIntreface` and `ImplementationImpl` things, e.g. [like this](https://howtodoinjava.com/java/basics/java-naming-conventions/)

File: src/test/java/com/incountry/StorageTest.java (hunk 4 above)
Comment (robert-par) [reply src/test/java/com/incountry/StorageTest.java]: puted for breakpoint and forgot delete

**Pull request merged**